core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -216,19 +223,26 @@ object ConsoleConsumer extends Logging {
     * are conflicting.
     */
   def setAutoOffsetResetValue(config: ConsumerConfig, props: Properties) {
+    val (earliestConfigName, latestConfigName) = if (config.useOldConsumer)
+      (OffsetRequest.SmallestTimeString, OffsetRequest.LargestTimeString)
+    else
+      ("earliest", "latest")
+
     if (props.containsKey(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)) {
       // auto.offset.reset parameter was specified on the command line
-      if (config.options.has(config.resetBeginningOpt) && "latest".equals(props.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG))) {
+      if (config.options.has(config.resetBeginningOpt) && earliestConfigName != props.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)) {
```

Comment:
Hmm.. I debated on this a little. If you pass the wrong config value (e.g. "latest" for the old consumer), the old check wouldn't catch it. We would still get an error when we construct the consumer, so maybe that's fine? Alternatively, we can change the message to refer to the overridden value.